pkg/secretless/plugin/connector/resources.go

```diff
@@ -0,0 +1,42 @@
+package connector
+
+import (
+	"github.com/cyberark/secretless-broker/pkg/secretless/log"
+)
+
+// Resources is an interface that defines what things will
+// be needed by a plugin at runtime.
+type Resources interface {
+	// Config returns the content of addition configuration
+	// parameters passed to the connector.
+	Config() []byte
+
+	// Logger returns an instance of a Logger that can be used
+	// to record logging messages from the plugin.
+	Logger() log.Logger
+}
+
+type resourcesHolder struct {
```

Comment:
is the stuff from here down meant to be here?

Reply:
Yeah. We cannot return an interface without a backing object that implements that interface, thus we have the `resourcesHolder`

Reply:
Just call this `resources` or if you're worried about confusion with `Resources`, append or prepend an `_` too.